adminapi/dataset.py

```diff
@@ -417,6 +417,14 @@ def _save_old_value(self, key):
                 self.old_values[key] = old_value
 
     def __setitem__(self, key, value):
+        if type(value) is str:
+            # We don't know the data type of the attribute here.
+            #
+            # If the attribute value is a string we try to match it against
+            # our known data types (True = 1, 0, true, false) etc. and convert
+            # it. If nothing matches we assume it is a string.
+            value = str_to_datatype(value)
```

Comment:
Is this really safe without taking the type of field into account? Isn't it possible to store `1` or `true` as a real string?

Reply:
In short: No it is not 100% safe but it "should" work for our setup.

In details:

We currently don't know the data type on the client side. If we would want to know the data type we would have to Query it at some point from the backend which can be quite expensive. I decided against this because it is a big change with huge impact.

I can happen for example that somebody creates a string attribute and does not properly set the regex validation and tries to store "false" in it.

I want to refactor this again later by either finding a cheap way to know the data type or find a another solution but this is out of the scope of this PR. Right now we can't commit  datetimes in the Servershell edit page at all.

Reply:
Ok, got it. Then good to go for now.